design/specs/add-formula-column.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Listing Views Spec
+description: 
+published: true
+date: 2022-06-20T08:24:04Z
+tags: 
+editor: markdown
+dateCreated: 2022-06-20T08:24:04Z
+---
+
+## Context
+
+In addition to data columns, we have formula columns as well. These columns are basically derived from a formula. A formula may or may not use another data column to generate the formula column. The formula can also use parameters of different data types other than columns. 
+
+[Wireframe link]()
+
+## Scenarios
+
+### Scenario 1 - Add a formula column without clicking outside of the inspector.
+
+On the left hand side of the explorer view, there is a  `Add Column`  button. When user clicks on it, there are two options visible to them. One is `Add direct column` and the other is `Add formula column`. To add a formula column user will need to click on `Add formula column`. 
+
+![](/assets/design/specs/add-formula-column/1.png)
+
+Once the user clicks on `Add formula column` option, the inspector shows formula settings that include `Name` input of the formula and a list of formulas to select from. The list of formulas is described in detail in the component section. 
+
+If the name is left empty by user for the next steps, there is a default name given to the column that is ColumnX. X is the number given to the column. For instance if its the first default named column it will be Column1.
+
+![](/assets/design/specs/add-formula-column/2.png)
```

Comment:
Can we move all wireframes to the top of this document and add them as balsamiq links instead of embedding them? It will make maintaining the document a lot easier.

Reply:
Sure.